model/gamut_compress.ctl

```diff
@@ -0,0 +1,135 @@
+
+// <ACEStransformID>urn:ampas:aces:transformId:v1.5:LMT.Academy.GamutCompress.a1.0.1</ACEStransformID>
+// <ACESuserName>ACES 1.3 Look - Gamut Compress</ACESuserName>
+
+//
+// "LMT" for compressing out of gamut scene-referred values into AP1 
+// 
+// Input and output are ACES2065-1
+//
+
+//
+// Direction:
+//    By default this transform operates in the forward direction (compressing the gamut).
+//    If instead an inverse operation is needed (undoing a prior gamut compression), there
+//    is a runtime flag to do this. In ctlrender, this can be achieved by appending 
+//    '-param1 invert 1' after the '-ctl gamut_compress.ctl' string.
+//
+
+
+
+import "ACESlib.Transform_Common";
+
+
+
+/* --- Gamut Compress Parameters --- */
+const float threshold_c = 0.815;
+const float threshold_m = 0.803;
+const float threshold_y = 0.880;
+const float power = 1.2;
+const float cyan =  0.147;
+const float magenta = 0.264;
+const float yellow = 0.312;
+
+
+
+// calculate compressed distance
+float compress(float dist, float lim, float thr, bool invert, float power)
+{
+    float cdist;
+    float s;
+    if (dist < thr) {
+        cdist = dist;
+    }
+    else { // power(p) compression function plot https://www.desmos.com/calculator/54aytu7hek
+        if (lim < 1.0001) {
+            return dist; // disable compression, avoid nan
+        }
+        s = (lim - thr) / pow(pow((1.0 - thr) / (lim - thr), -power) - 1.0, 1.0 / power); // calc y=1 intersect
+        if (!invert) {
+            cdist = thr + s * ((dist - thr) / s) / (pow(1.0 + pow((dist - thr) / s, power), 1.0 / power)); // compress
+        }
+        else {
+            if (dist > (thr + s)) {
+                cdist = dist; // avoid singularity
+            }
+            else {
+                cdist = thr + s * pow(-(pow((dist - thr) / s, power) / (pow((dist - thr) / s, power) - 1.0)), 1.0 / power); // uncompress
+            }
+        }
+    }
+    return cdist;
+}
+
+
+
+void main 
+(
+    input varying float rIn, 
+    input varying float gIn, 
+    input varying float bIn, 
+    input varying float aIn,
+    output varying float rOut,
+    output varying float gOut,
+    output varying float bOut,
+    output varying float aOut,
+    input uniform bool invert = false
+) 
+{ 
+    // source pixels
+    float ACES[3] = {rIn, gIn, bIn};
+
+    // convert to ACEScg
+    float lin_AP1[3] = mult_f3_f44(ACES, AP0_2_AP1_MAT);
+
+    // thr is the percentage of the core gamut to protect.
+    float thr[3] = {
+        min(0.9999, threshold_c),
+        min(0.9999, threshold_m),
+        min(0.9999, threshold_y)
+    };
```

Comment:
Since the threshold values are constants in this version there is no reason for this check.

Reply:
Agreed. I have pushed a commit which removes code which is redundant when the parameters are constants.